### PR TITLE
mgr/dashboard: fix indefinite loop in cephadm dashboard e2e

### PR DESCRIPTION
**Pull request by nizamial09**
the tests seems waiting to fetch the prometheus details incase the cephadm ran into error and it just waits there for more than an hour without any progress. fixing that and some minor improvements.

an example log: https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12287/consoleFull

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh [hunk 1]
```diff
@@ -82,3 +82,33 @@ while [[ -z $(kcli ssh -u root -- ceph-node-00 'journalctl --no-tail --no-pager
...
     kcli ssh -u root -- ceph-node-00 'journalctl -n 100 --no-pager -t cloud-init'
 done
+
+kcli ssh -u root ceph-node-00 'cephadm shell "ceph config set mgr mgr/prometheus/exclude_perf_counters false"'
+
+# check if the prometheus daemon is running on jenkins node
+# before starting the e2e tests
+if [[ -n "${JENKINS_HOME}" ]]; then
+    retry=0
+    PROMETHEUS_RUNNING_COUNT=$(kcli ssh -u root ceph-node-00 'cephadm shell "ceph orch ls --service_name=prometheus --format=json"' | jq -r '.[] | .status.running')
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh:92]: We can create a function for this and call, as we are using it multiple times.

**Review by afreen23 — CHANGES_REQUESTED** [review]
Thanks @nizamial09 for this.

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh]: done, thanks

**Review by afreen23 — APPROVED** [review]

**Comment by afreen23** [comment]
cephadm test failures:
```
[Test Result](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/) (22 failures / +11)
[RGW Daemon Overall Performance (example #1).Grafana panels RGW Daemon Overall Performance (example #1)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_1)/Grafana_panels_RGW_Daemon_Overall_Performance__example__1_/)
[RGW Daemon Overall Performance (example #2).Grafana panels RGW Daemon Overall Performance (example #2)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_2)/Grafana_panels_RGW_Daemon_Overall_Performance__example__2_/)
[RGW Daemon Overall Performance (example #3).Grafana panels RGW Daemon Overall Performance (example #3)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_3)/Grafana_panels_RGW_Daemon_Overall_Performance__example__3_/)
[RGW Daemon Overall Performance (example #4).Grafana panels RGW Daemon Overall Performance (example #4)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_4)/Grafana_panels_RGW_Daemon_Overall_Performance__example__4_/)
[RGW Daemon Overall Performance (example #5).Grafana panels RGW Daemon Overall Performance (example #5)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_5)/Grafana_panels_RGW_Daemon_Overall_Performance__example__5_/)
[RGW Daemon Overall Performance (example #6).Grafana panels RGW Daemon Overall Performance (example #6)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_6)/Grafana_panels_RGW_Daemon_Overall_Performance__example__6_/)
[RGW per Daemon Performance (example #1).Grafana panels RGW per Daemon Performance (example #1)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_1)/Grafana_panels_RGW_per_Daemon_Performance__example__1_/)
[RGW per Daemon Performance (example #2).Grafana panels RGW per Daemon Performance (example #2)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_2)/Grafana_panels_RGW_per_Daemon_Performance__example__2_/)
[RGW per Daemon Performance (example #3).Grafana panels RGW per Daemon Performance (example #3)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_3)/Grafana_panels_RGW_per_Daemon_Performance__example__3_/)
[RGW per Daemon Performance (example #4).Grafana panels RGW per Daemon Performance (example #4)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_4)/Grafana_panels_RGW_per_Daemon_Performance__example__4_/)
[RGW per Daemon Performance (example #5).Grafana panels RGW per Daemon Performance (example #5)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_5)/Grafana_panels_RGW_per_Daemon_Performance__example__5_/)
[RGW per Daemon Performance (example #6).Grafana panels RGW per Daemon Performance (example #6)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_6)/Grafana_panels_RGW_per_Daemon_Performance__example__6_/)
[should check if mon daemon is running on all hosts.when cluster creation is completed Hosts page should check if mon daemon is running on all hosts](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/should%20check%20if%20mon%20daemon%20is%20running%20on%20all%20hosts/when_cluster_creation_is_completed_Hosts_page_should_check_if_mon_daemon_is_running_on_all_hosts/)
[should create an mds service.Services page should create an mds service](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/should%20create%20an%20mds%20service/Services_page_should_create_an_mds_service/)
[should stop a daemon.Services page should stop a daemon](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/should%20stop%20a%20daemon/Services_page_should_stop_a_daemon/)
[should create and delete snmp-gateway service with version V3.Services page should create and delete snmp-gateway service with version V3](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/should%20create%20and%20delete%20snmp-gateway%20service%20with%20version%20V3/Services_page_should_create_and_delete_snmp_gateway_service_with_version_V3/)
[should create an NFS cluster.nfsExport page Create, edit and delete should create an NFS cluster](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/should%20create%20an%20NFS%20cluster/nfsExport_page_Create__edit_and_delete_should_create_an_NFS_cluster/)
[should check if rgw daemon is running on all hosts.Host Page should check if rgw daemon is running on all hosts](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/should%20check%20if%20rgw%20daemon%20is%20running%20on%20all%20hosts/Host_Page_should_check_if_rgw_daemon_is_running_on_all_hosts/)
[should force maintenance and exit.Host Page should force maintenance and exit](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/should%20force%20maintenance%20and%20exit/Host_Page_should_force_maintenance_and_exit/)
[should edit a service.Create cluster create services page when Orchestrator is available should edit a service](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/should%20edit%20a%20service/Create_cluster_create_services_page_when_Orchestrator_is_available_should_edit_a_service/)
[Add and remove labels on host (example #1).Cluster expansion host addition Add and remove labels on host (example #1)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/Add%20and%20remove%20labels%20on%20host%20(example%20_1)/Cluster_expansion_host_addition_Add_and_remove_labels_on_host__example__1_/)
[should create ingress as unmanaged.Services page should create ingress as unmanaged](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12351/testReport/junit/(root)/should%20create%20ingress%20as%20unmanaged/Services_page_should_create_ingress_as_unmanaged/)
```

dashboard test failures:

```
[Test Result](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/) (10 failures / +3)
[should delete image.Images page create, edit & delete image test should delete image](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/should%20delete%20image/Images_page_create__edit___delete_image_test_should_delete_image/)
[should move the image to the trash.Images page move to trash, restore and purge image tests should move the image to the trash](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/should%20move%20the%20image%20to%20the%20trash/Images_page_move_to_trash__restore_and_purge_image_tests_should_move_the_image_to_the_trash/)
[should restore image to images table.Images page move to trash, restore and purge image tests should restore image to images table](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/should%20restore%20image%20to%20images%20table/Images_page_move_to_trash__restore_and_purge_image_tests_should_restore_image_to_images_table/)
[should purge trash in images trash tab.Images page move to trash, restore and purge image tests should purge trash in images trash tab](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/should%20purge%20trash%20in%20images%20trash%20tab/Images_page_move_to_trash__restore_and_purge_image_tests_should_purge_trash_in_images_trash_tab/)
["after all" hook for "should purge trash in images trash tab".Images page move to trash, restore and purge image tests "after all" hook for "should purge trash in images trash tab"](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/%22after%20all%22%20hook%20for%20%22should%20purge%20trash%20in%20images%20trash%20tab%22/Images_page_move_to_trash__restore_and_purge_image_tests__after_all__hook_for__should_purge_trash_in_images_trash_tab_/)
["after each" hook for "should show the correct text for the tab labels".OSDs page check existence of fields on OSD page by selecting one row in OSDs List "after each" hook for "should show the correct text for the tab labels"](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/%22after%20each%22%20hook%20for%20%22should%20show%20the%20correct%20text%20for%20the%20tab%20labels%22/OSDs_page_check_existence_of_fields_on_OSD_page_by_selecting_one_row_in_OSDs_List__after_each__hook_for__should_show_the_correct_text_for_the_tab_labels_/)
[should switch to the second cluster and back to hub.Muti-cluster management page should switch to the second cluster and back to hub](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/should%20switch%20to%20the%20second%20cluster%20and%20back%20to%20hub/Muti_cluster_management_page_should_switch_to_the_second_cluster_and_back_to_hub/)
[should reconnect the second cluster.Muti-cluster management page should reconnect the second cluster](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/should%20reconnect%20the%20second%20cluster/Muti_cluster_management_page_should_reconnect_the_second_cluster/)
[should edit the second cluster.Muti-cluster management page should edit the second cluster](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/should%20edit%20the%20second%20cluster/Muti_cluster_management_page_should_edit_the_second_cluster/)
[should disconnect the second cluster.Muti-cluster management page should disconnect the second cluster](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/16233/testReport/junit/(root)/should%20disconnect%20the%20second%20cluster/Muti_cluster_management_page_should_disconnect_the_second_cluster/)
```

**Comment by afreen23** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
[The cephadm failure](https://github.com/ceph/ceph/pull/59691#issuecomment-2348571815) not seen after re-run

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
In cephadm tests some docker enc problem happening seen on this PR,[ this PR](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12444/console).

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
new failures for cephadm

```
[Test Result](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/) (22 failures / +17)
[should check if rgw daemon is running on all hosts.Host Page should check if rgw daemon is running on all hosts](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/should%20check%20if%20rgw%20daemon%20is%20running%20on%20all%20hosts/Host_Page_should_check_if_rgw_daemon_is_running_on_all_hosts/)
[should force maintenance and exit.Host Page should force maintenance and exit](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/should%20force%20maintenance%20and%20exit/Host_Page_should_force_maintenance_and_exit/)
[should drain, remove and add the host back.Host Page should drain, remove and add the host back](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/should%20drain,%20remove%20and%20add%20the%20host%20back/Host_Page_should_drain__remove_and_add_the_host_back/)
[should create and delete snmp-gateway service with version V3 and w/o privacy protocol.Services page should create and delete snmp-gateway service with version V3 and w/o privacy protocol](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/should%20create%20and%20delete%20snmp-gateway%20service%20with%20version%20V3%20and%20w_o%20privacy%20protocol/Services_page_should_create_and_delete_snmp_gateway_service_with_version_V3_and_w_o_privacy_protocol/)
[RGW Daemon Overall Performance (example #1).Grafana panels RGW Daemon Overall Performance (example #1)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_1)/Grafana_panels_RGW_Daemon_Overall_Performance__example__1_/)
[RGW Daemon Overall Performance (example #2).Grafana panels RGW Daemon Overall Performance (example #2)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_2)/Grafana_panels_RGW_Daemon_Overall_Performance__example__2_/)
[RGW Daemon Overall Performance (example #3).Grafana panels RGW Daemon Overall Performance (example #3)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_3)/Grafana_panels_RGW_Daemon_Overall_Performance__example__3_/)
[RGW Daemon Overall Performance (example #4).Grafana panels RGW Daemon Overall Performance (example #4)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_4)/Grafana_panels_RGW_Daemon_Overall_Performance__example__4_/)
[RGW Daemon Overall Performance (example #5).Grafana panels RGW Daemon Overall Performance (example #5)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_5)/Grafana_panels_RGW_Daemon_Overall_Performance__example__5_/)
[RGW Daemon Overall Performance (example #6).Grafana panels RGW Daemon Overall Performance (example #6)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_6)/Grafana_panels_RGW_Daemon_Overall_Performance__example__6_/)
[RGW per Daemon Performance (example #1).Grafana panels RGW per Daemon Performance (example #1)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_1)/Grafana_panels_RGW_per_Daemon_Performance__example__1_/)
[RGW per Daemon Performance (example #2).Grafana panels RGW per Daemon Performance (example #2)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_2)/Grafana_panels_RGW_per_Daemon_Performance__example__2_/)
[RGW per Daemon Performance (example #3).Grafana panels RGW per Daemon Performance (example #3)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_3)/Grafana_panels_RGW_per_Daemon_Performance__example__3_/)
[RGW per Daemon Performance (example #4).Grafana panels RGW per Daemon Performance (example #4)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_4)/Grafana_panels_RGW_per_Daemon_Performance__example__4_/)
[RGW per Daemon Performance (example #5).Grafana panels RGW per Daemon Performance (example #5)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_5)/Grafana_panels_RGW_per_Daemon_Performance__example__5_/)
[RGW per Daemon Performance (example #6).Grafana panels RGW per Daemon Performance (example #6)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/RGW%20per%20Daemon%20Performance%20(example%20_6)/Grafana_panels_RGW_per_Daemon_Performance__example__6_/)
[should check if mon daemon is running on all hosts.when cluster creation is completed Hosts page should check if mon daemon is running on all hosts](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/should%20check%20if%20mon%20daemon%20is%20running%20on%20all%20hosts/when_cluster_creation_is_completed_Hosts_page_should_check_if_mon_daemon_is_running_on_all_hosts/)
[should create a nfs-export with RGW backend.nfsExport page Create, edit and delete should create a nfs-export with RGW backend](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/should%20create%20a%20nfs-export%20with%20RGW%20backend/nfsExport_page_Create__edit_and_delete_should_create_a_nfs_export_with_RGW_backend/)
[should show Clients.nfsExport page Create, edit and delete should show Clients](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/should%20show%20Clients/nfsExport_page_Create__edit_and_delete_should_show_Clients/)
[should edit an export.nfsExport page Create, edit and delete should edit an export](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/should%20edit%20an%20export/nfsExport_page_Create__edit_and_delete_should_edit_an_export/)
[should delete exports and bucket.nfsExport page Create, edit and delete should delete exports and bucket](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/should%20delete%20exports%20and%20bucket/nfsExport_page_Create__edit_and_delete_should_delete_exports_and_bucket/)
[Add and remove labels on host (example #1).Cluster expansion host addition Add and remove labels on host (example #1)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/12454/testReport/junit/(root)/Add%20and%20remove%20labels%20on%20host%20(example%20_1)/Cluster_expansion_host_addition_Add_and_remove_labels_on_host__example__1_/)
```

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by nizamial09** [comment]
merging this so that we cephadm e2e won't get stuck in a loop atleast.

**Pull request merged**